sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/services/OpenAiServiceV2.java

```diff
@@ -102,64 +92,36 @@ public OpenAiChatCompletionResponse chatCompletionImage(@Nonnull final String li
   @Nonnull
   public OpenAiChatCompletionResponse chatCompletionToolExecution(
       @Nonnull final String location, @Nonnull final String unit) {
-
-    // 1. Define the function
-    final Map<String, Object> schemaMap = generateSchema(WeatherMethod.Request.class);
-    final var function =
-        new FunctionObject()
-            .name("weather")
-            .description("Get the weather for the given location")
-            .parameters(schemaMap);
-    final var tool = new ChatCompletionTool().type(FUNCTION).function(function);
-
     final var messages = new ArrayList<OpenAiMessage>();
     messages.add(OpenAiMessage.user("What's the weather in %s in %s?".formatted(location, unit)));
 
-    // Assistant will call the function
-    final var request = new OpenAiChatCompletionRequest(messages).withTools(List.of(tool));
+    // 1. Define the function
+    final var weatherFunction =
+        new OpenAiFunctionTool("weather", WeatherMethod.Request.class)
+            .withDescription("Get the weather for the given location");
```

Comment:
The is no use-case for `OpenAiFunctionTool` to be immutable.
Also, is there ever going to be another `OpenAiTool` that is not `OpenAiFunctionTool`?